File: POH7/SantaFuku/Main.js
// Try POH
// author: Leonardone @ NEETSDKASU
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function(c){main(c);});

function main(chunk) {
    var i, d, a;
    var s = chunk.toString().trim().split(/\s/).map(function(c,i,a){return parseInt(c);});
    var x = s[0], y = s[1], z = s[2], n = s[3];
    var xs = [], ys = [], xm = x, ym = y;
    var f = function (es, em, e, b) {
        var i;
        em = Math.min(em, e, b - e);
        for (i in es) {
            em = Math.min(em, Math.abs(es[i] - e));
        }
        es.push(e);
        return em;
    };
    for (i = 0; i < n; i++) {
        d = s[i * 2 + 4];
        a = s[i * 2 + 5];
        if (d === 0) {
            xm = f(xs, xm, a, x);
        } else {
            ym = f(ys, ym, a, y);
        }
    }
    console.log(z * xm * ym);
}

File: POH7/SantaFuku/Main.java
// Try POH
// author: Leonardone @ NEETSDKASU
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(System.in);
        
        int x = in.nextInt();
        int y = in.nextInt();
        int z = in.nextInt();
        int n = in.nextInt();
        
        int[][] xy = new int[2][n];
        int[] xyc = {0, 0};
        
        int minxy[] = {x, y};
        
        for (int i = 0; i < n; i++) {
            int d = in.nextInt();
            int a = in.nextInt();
            minxy[d] = Math.min(minxy[d], Math.min(a, (d == 0 ? x : y) - a));
            for (int j = 0; j < xyc[d]; j++) {
                minxy[d] = Math.min(minxy[d], Math.abs(xy[d][j] - a));
            }
            xy[d][xyc[d]] = a;
            xyc[d]++;
        }
        
        System.out.println(z * minxy[0] * minxy[1]);
        
    }
}
